src/core/server/elasticsearch/types.ts

```diff
@@ -34,11 +34,17 @@ import { ServiceStatus } from '../status';
  * @public
  */
 export interface ElasticsearchServiceSetup {
+  /**
+   * Provide direct access to the current elasticsearch configuration.
+   *
+   * @deprecated this will be removed in a later version.
+   */
+  readonly config$: Observable<ElasticsearchConfig>;
```

Comment:
- I could have kept `legacy.config$` and expose it on the public contract instead, but now that this is going to be consumed by KP plugins, I thought it was better on top level. Also, that would allow to totally get rid of `esSetup.legacy` when we remove the legacy client

- flagged it as deprecated. Not sure if it's the right move.

Reply:
I'd prefer to keep it under `legacy` namespace to make it even more explicit that we are going to remove it. @joshdover any thoughts?

Reply:
Deprecation tags are easy to miss (they're really only present in intellisense right now since we don't have any linter warnings for using deprecated APIs). Given that, I also prefer keeping it under the `legacy` namespace to make it clear that this will be removed in the future.

Reply:
Related, this new feature in TS v4 / VSCode will make deprecated APIs more apparent: https://devblogs.microsoft.com/typescript/announcing-typescript-4-0-rc/#deprecated-support